imgui.lua

```diff
@@ -46,11 +47,23 @@ imgui.Hook("PreRender", "Input", function()
 	-- calculate mouse state
 	if shouldAcceptInput() then
 		local wasPressing = gState.pressing
-		gState.pressing = input.IsMouseDown(MOUSE_LEFT) or input.IsKeyDown(KEY_E)
+		gState.pressing = pressing[IN_USE] or pressing[IN_ATTACK]
 		gState.pressed = not wasPressing and gState.pressing
 	end
 end)
 
+hook.Add("KeyPress", "imgui_keypress", function(ply, key)
```

Comment:
This hook might get overridden by another imgui instance (this has actually happened). There's the `imgui.Hook` helper you could use to make per-instance hook.

Reply:
I wasn't aware of that, thanks for the tip, I actually found a more robust yet slightly less efficient solution than the original implementation. I also added an :Alive() check for my own personal usage.
```lua
imgui.Hook("PreRender", "Input", function()
	-- calculate mouse state
	if shouldAcceptInput() then
		local ply = LocalPlayer()
		local wasPressing = gState.pressing
		gState.pressing = ply:Alive() and (ply:KeyDown(IN_USE) or ply:KeyDown(IN_ATTACK))
		gState.pressed = not wasPressing and gState.pressing
	end
end)
```
Would you prefer using `imgui.Hook` with my previous method or this new method?

I've also implemented a distanceInput parameter to limit the distance in which a player can interact with the interface if you'd like me to make a pull request for that.

Reply:
> Would you prefer using imgui.Hook with my previous method or this new method?
Hm, this new method seems good to me. Not sure about the alive check, since that seems like something that would be better handled on application or server level.

I'll need to think about the distance input thing as well, since the lines between api complexity and relative usefulness start getting a bit blurry. You can of course make a PR to discuss all that in if you have the code ready though.

Reply:
I'll omit the Alive check and push these changes to master. I'll open up a new pull request in a bit to discuss the other feature and how I implemented it.